django/db/models/fields/tuple_lookups.py

```diff
@@ -101,6 +133,10 @@ class TupleExact(TupleLookupMixin, Exact):
     def as_oracle(self, compiler, connection):
         # Process right-hand-side to trigger sanitization.
         self.process_rhs(compiler, connection)
+        if isinstance(self.rhs, Subquery):
+            raise NotImplementedError(
+                "Subquerying on composite fields is not yet implemented on Oracle."
+            )
```

Comment:
I would suggest defining a custom error class with this message to re-use below.